myriad-commons/src/main/java/com/ebay/myriad/common/FrameworkUpdate.java

```diff
@@ -0,0 +1,51 @@
+package com.ebay.myriad.common;
+
+import com.google.gson.Gson;
+
+/**
+ * 
+ * Framework update from the scheduler to the executor
+ *
+ */
+public class FrameworkUpdate {
+    /**
+     * Define what kind of this update is.
+     */
+    public static enum Type {Command};
+    
+    /**
+     * Define what kind of operation or command this update is.
+     */
+    public static enum Operation {Shutdown};
+    
+    public FrameworkUpdate() {
+    }
+    
+    public FrameworkUpdate(Type type, Operation operation) {
+        setType(type);
+        setOperation(operation);
+    }
+    
+    public Type getType() {
+        return type;
+    }
+    
+    public void setType(Type type) {
+        this.type = type;
+    }
+    
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+    
+    public Operation getOperation() {
+        return operation;
+    }
+    
+    public String toString () {
+        return new Gson().toJson(this);
+    }
+    
+    private Type type;
+    private Operation operation;
+}
```

Comment:
The idea was/is that additional enumerated values could/would be added
later. As to Open/Close, no matter what class the enumerated types are
defined in that class will need changed, unless at this time all of the
possible enumerated values can be determined which seems unrealistic.  Of
course if you feel that moving the definitions to a "Constants" class would
mitigate the Open/close issue, that constanst classes do not follow oop I
have no problem with that although the two types are intimatedly tied to
this class.

On Wed, Aug 12, 2015 at 9:43 AM, Ken Sipe <notifications@github.com> wrote:

> In
> myriad-commons/src/main/java/com/ebay/myriad/common/FrameworkUpdate.java
> <https://github.com/mesos/myriad/pull/115#discussion_r36860323>:
>
> > +
> > +    public void setOperation(Operation operation) {
> > +        this.operation = operation;
> > +    }
> > +
> > +    public Operation getOperation() {
> > +        return operation;
> > +    }
> > +
> > +    public String toString () {
> > +        return new Gson().toJson(this);
> > +    }
> > +
> > +    private Type type;
> > +    private Operation operation;
> > +}
>
> there are several things that feel odd about this case... the first is
> there are 2 enums which only have 1 defined possibility. The second is, if
> we assume that there will be more Types and Operations, that they are
> defined inside of this class. This violates good OOP practice of the
> open/close principle.
>
> thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/myriad/pull/115/files#r36860323>.
>